snapcraft/storeapi/v2/releases.py

```diff
@@ -0,0 +1,180 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, List, Optional
+
+import jsonschema
+
+from ._api_schema import RELEASES_JSONSCHEMA
+
+"""
+This module holds representations for results for the v2 releases
+API endpoint provided by the Snap Store. It currently only covers
+Revisions and Releases.
+
+The full API is documented on
+https://dashboard.snapcraft.io/docs/v2/en/snaps.html#snap-releases
+"""
+
+
+class Release:
+    """Represent a snap release."""
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Any]) -> "Release":
+        jsonschema.validate(
+            payload,
+            RELEASES_JSONSCHEMA["properties"]["releases"]["items"]["properties"],
+        )
+        return cls(
+            architecture=payload["architecture"],
+            branch=payload["branch"],
+            channel=payload["channel"],
+            expiration_date=payload["expiration-date"],
+            revision=payload["revision"],
+            risk=payload["risk"],
+            track=payload["track"],
+            when=payload["when"],
+        )
+
+    def marshal(self) -> Dict[str, Any]:
+        return {
+            "architecture": self.architecture,
+            "branch": self.branch,
+            "channel": self.channel,
+            "expiration-date": self.expiration_date,
+            "revision": self.revision,
+            "risk": self.risk,
+            "track": self.track,
+            "when": self.when,
+        }
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__}: {self.channel!r}>"
+
+    def __init__(
+        self,
+        architecture: str,
+        branch: Optional[str],
+        channel: str,
+        expiration_date: Optional[str],
+        revision: int,
+        risk: str,
+        track: str,
+        when: str,
+    ) -> None:
+        self.architecture = architecture
+        self.branch = branch
+        self.channel = channel
+        self.expiration_date = expiration_date
+        self.revision = revision
+        self.risk = risk
+        self.track = track
+        self.when = when
+
+
+class Revision:
+    """Represent a snap revision."""
+
+    @classmethod
+    def unmarshal(cls, payload: Dict[str, Any]) -> "Revision":
+        jsonschema.validate(
+            payload,
+            RELEASES_JSONSCHEMA["properties"]["revisions"]["items"]["properties"],
+        )
+        return cls(
+            architectures=payload["architectures"],
+            base=payload["base"],
+            build_url=payload["build_url"],
+            confinement=payload["confinement"],
+            created_at=payload["created_at"],
+            grade=payload["grade"],
+            revision=payload["revision"],
```

Comment:
"revision" is not a required field in the schema.

Reply:
this has to be a mistake

Reply:
I fixed the schema,
```
                "required": [
                    "architectures",
                    "attributes",
                    "base",
                    "build_url",
                    "confinement",
                    "created_at",
                    "epoch",
                    "grade",
                    "required",
                    "sha3-384",
                    "size",
                    "status",
                    "version"
                ],
```

"required" is supposed to be "revision"